Task 2

ARybakova opened the pull request:
there is a problem with search functionality

ARybakova committed:
Task 2

aspirisen commented on server/server.js:
```diff
@@ -1,10 +1,14 @@
 
+const static_path = path.join(__dirname, './../built');
 
 app.use(express.static('built'));
```
Было бы хорошо встроить webpack-dev-server в твой скрипт чтобы за один раз все билдилось и поднимался сервер

aspirisen commented on src/components/App.jsx:
```diff
@@ -0,0 +1,16 @@
+import { MoviePage } from './MoviePage.jsx';
+
+const App = () => (
+    //<MoviePage/>
```
Не нужно коммитить закоменченный код

aspirisen commented on src/components/Movie.jsx:
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+export const Movie = ( {movie} ) => (
```
обычно записывают так 
`export const Movie = ({ movie }) => (`

aspirisen commented on src/components/Results.jsx:
```diff
@@ -0,0 +1,72 @@
+        super(props);
+        this.state = {
+            sortMethod: "sortByRelease",
+            data: this.props.data.sort((a,b) => b.release_year - a.release_year )
```
`sort` это мутирующий метод, перед ним нужно сделать `slice`

aspirisen commented on src/components/Results.jsx:
```diff
@@ -0,0 +1,72 @@
+
+                { this.state.data.length > 0 ? (
+
+                    <div>
```
Я бы вынес из в два отдельных метода чтобы это условие было более наглядное

aspirisen commented on src/components/Results.jsx:
```diff
@@ -0,0 +1,72 @@
+                            <span className="results_title">{this.state.data.length} movies found</span>
+                            <div className="results_filter">
+                                <span className="results_filter_title">Sort by</span>
+                                <button className={'results_filter_btn' + (this.state.sortMethod === "sortByRelease" ? ' active' : '')} onClick={this.sortByRelease.bind(this)}>release date</button>
```
если много или длинные пропсы то тх записывают в столбик

ARybakova committed:
Task 2:
1. Created different files for content of SearchPage and MoviePage
2. Fixed bug in sorting methods ('slice' added)
3. Fixed some 'cosmetic' bugs

aspirisen commented:
Еще нужно добавить временные файлы и node_module в .gitignore

ARybakova committed:
Task 2:
1. added webpack-dev-middleware
2. added gitignore
